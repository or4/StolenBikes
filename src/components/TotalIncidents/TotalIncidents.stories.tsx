import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, number } from '@storybook/addon-knobs';

import { TotalIncidents } from './TotalIncidents';

const stories = storiesOf('TotalIncidents', module);
stories.addDecorator(withKnobs);

stories.add('TotalIncidents', () => {
    return <TotalIncidents totalIncidents={getKnobsTotalIncidents()} />;
});

function getKnobsTotalIncidents() {
    return number('Total incidents', 18);
}
