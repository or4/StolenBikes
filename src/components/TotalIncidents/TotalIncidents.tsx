import React from 'react';
import styled from 'styled-components';

export interface IProps {
    totalIncidents: number;
}

const Container = styled.div`
    min-height: 40px;
    line-height: 40px;
    text-align: right;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
`;

export const TotalIncidents: React.FC<IProps> = ({ totalIncidents }: IProps) => (
    <Container>{`Total: ${totalIncidents}`}</Container>
);
