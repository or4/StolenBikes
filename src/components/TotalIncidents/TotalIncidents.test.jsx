import React from 'react';
import { mount } from 'enzyme';

import { TotalIncidents } from './TotalIncidents';

describe('TotalIncidents', () => {
    it('should render correctly', () => {
        const component = <TotalIncidents totalIncidents={18} />;

        const container = mount(component);
        expect(container).toMatchSnapshot();
        container.unmount();
    });
});
