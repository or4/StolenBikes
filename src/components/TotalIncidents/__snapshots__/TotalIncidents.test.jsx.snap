// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TotalIncidents should render correctly 1`] = `
<TotalIncidents
  totalIncidents={18}
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "fKVUqC",
            "rules": Array [
              "
    min-height: 40px;
    line-height: 40px;
    text-align: right;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bdVaJa fKVUqC"
      >
        Total: 18
      </div>
    </StyledComponent>
  </styled.div>
</TotalIncidents>
`;
